### PR TITLE
Add ActiveMQ input plugin

### LINKED ISSUE #2622: Plugin suggestion: ActiveMQ
Plugin support to push activemq status like no of topics, topic wise consumer count, queue wise enqueued and dequeued messages count can be pushed to influxdb for monitoring.

[ActiveMQ Doc](http://activemq.apache.org/getting-started.html)

### PR DESCRIPTION
**Pull request by mlabouardy**
Resolves #2622
### Required for all PRs:

- [x] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

### CODE REVIEW

File: plugins/inputs/activemq/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - selector
+    - active
+
+### Example Output:
+
+```
+$ ./telegraf -config telegraf.conf -input-filter activemq -test
+queues_metrics,name=sandra,host=88284b2fe51b consumer_count=0i,enqueue_count=0i,dequeue_count=0i,size=0i 1492610703000000000
+queues_metrics,name=Test,host=88284b2fe51b dequeue_count=0i,size=0i,consumer_count=0i,enqueue_count=0i 1492610703000000000
+topics_metrics,name=ActiveMQ.Advisory.MasterBroker\ ,host=88284b2fe51b size=0i,consumer_count=0i,enqueue_count=1i,dequeue_count=0i 1492610703000000000
```
Comment (danielnelson) [inline plugins/inputs/activemq/README.md]: Seems like there might be a bug where the name has trailing whitespace characters.

File: plugins/inputs/activemq/activemq.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+	## Required ActiveMQ Endpoint
+	# server = "192.168.50.10"
+	## Required ActiveMQ port
+	# port = 8161
+	## Required username used for request HTTP Basic Authentication
+	# username = "admin"
+	## Required password used for HTTP Basic Authentication
+	# password = "admin"
+	## Required ActiveMQ webadmin root path
+	# webadmin = "admin"
```
Comment (danielnelson) [inline plugins/inputs/activemq/activemq.go]: Do a two space indent here

File: plugins/inputs/activemq/activemq.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+	}
+
+	a.GatherQueuesMetrics(acc, queues)
+
+	a.GatherTopicsMetrics(acc, topics)
+
+	a.GatherSubscribersMetrics(acc, subscribers)
+
+	return nil
+}
```
Comment (danielnelson) [inline plugins/inputs/activemq/activemq.go]: Can you reduce the amount of vertical whitespace in this function?

File: plugins/inputs/activemq/activemq.go [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+	resp, err := client.Do(req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	return ioutil.ReadAll(resp.Body)
+}
```
Comment (danielnelson) [inline plugins/inputs/activemq/activemq.go]: Can you reduce the amount of vertical whitespace in this function?

**Comment by danielnelson** [comment]
@mlabouardy I think the tests will pass if we rebase the branch, could you do that for me and then I'll take a final look.

**Comment by mlabouardy** [comment]
@danielnelson done

**Comment by mlabouardy** [comment]
@danielnelson I cannot make the test pass, any idea what I did wrong ?

File: plugins/inputs/activemq/activemq.go [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+
+func (a *ActiveMQ) GatherTopicsMetrics(acc telegraf.Accumulator, topics Topics) {
+	for _, topic := range topics.TopicItems {
+		records := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		tags["name"] = topic.Name
+		tags["source"] = a.Server
+		tags["port"] = string(a.Port)
```
Comment (danielnelson) [inline plugins/inputs/activemq/activemq.go]: This won't work, instead use `strconv.FormatInt`.

File: plugins/inputs/activemq/activemq_test.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+	tags["port"] = "8161"
+
+	records["size"] = 0
+	records["consumer_count"] = 0
+	records["enqueue_count"] = 0
+	records["dequeue_count"] = 0
+
+	var acc testutil.Accumulator
+
+	activeMQ := new(ActiveMQ)
```
Comment (danielnelson) [inline plugins/inputs/activemq/activemq_test.go]: You will need to initialize the struct in the same way as in the init function:
```
activeMQ := &ActiveMQ{
	Server: "localhost",
	Port:   8161,
}
```

File: plugins/inputs/activemq/activemq.go [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+const (
+	QUEUES_STATS      = "queues"
+	TOPICS_STATS      = "topics"
+	SUBSCRIBERS_STATS = "subscribers"
+)
+
+var sampleConfig = `
+  ## Required ActiveMQ Endpoint
+  # server = "192.168.50.10"
+  ## Required ActiveMQ port
```
Comment (glinton) [inline plugins/inputs/activemq/activemq.go]: Nitpick/suggestion: add newlines between config options

Comment (mlabouardy) [reply plugins/inputs/activemq/activemq.go]: done

**Comment by mlabouardy** [comment]
@danielnelson when this can be merged ? :)

**Pull request merged**